graph/src/data/query/query.rs

```diff
@@ -134,6 +135,7 @@ pub struct Query {
     pub document: q::Document,
     pub variables: Option<QueryVariables>,
     pub shape_hash: u64,
+    pub hash: u64,
```

Comment:
Can we call this field something else or at least have a comment what the hash is for? I'd suggest `validation_hash` to make it clear that this is a hash we calculate for validations.